cellmlmanip/parser.py

```diff
@@ -68,22 +68,19 @@ def __init__(self, name):
 
     def set_parent(self, parent_name):
         """Sets the parent of this component"""
-        if self.parent:
-            raise ValueError('Parent of component %s already %s. Cannot set %s!' % (self.name,
-                                                                                    self.parent,
-                                                                                    parent_name))
+        assert not self.parent, 'Parent of component %s already %s. Cannot set %s!' % (self.name, self.parent,
+                                                                                       parent_name)
         self.parent = parent_name
 
     def add_sibling(self, sibling_name):
         """Adds a sibling for this component"""
-        if sibling_name in self.siblings:
-            raise ValueError('Sibling component %s already added!' % sibling_name)
+        assert sibling_name not in self.siblings, 'Sibling component %s already added!' % sibling_name
         self.siblings.add(sibling_name)
 
     def add_encapsulated(self, encapsulated_name):
         """Adds an encapsulated component to this component"""
-        if encapsulated_name in self.encapsulated:
-            raise ValueError('Encapsulated component %s already added!' % encapsulated_name)
+        assert encapsulated_name not in self.encapsulated, 'Encapsulated component %s already added!' % \
```

Comment:
I really can't come up with a test for this since the `_Component` class is private and I don't think it should ever be possible, hence I turned it into an assert (with the added benefit that it doesn't count as a coverage miss). Let me know what you think.

Reply:
I guess it was trying to capture the case where the CellML specifies the same encapsulation relationship twice?
If so, you should be able to trigger it with something like

```
   <group>
      <relationship_ref relationship="encapsulation"/>
      <component_ref component="sodium_current">
         <component_ref component="sodium_current_m_gate"/>
         <component_ref component="sodium_current_m_gate"/>
      </component_ref>
   </group>
```
and/or
```
   <group>
      <relationship_ref relationship="encapsulation"/>
      <component_ref component="sodium_current">
         <component_ref component="sodium_current_m_gate"/>
      </component_ref>
   </group>
   <group>
      <relationship_ref relationship="encapsulation"/>
      <component_ref component="sodium_current">
         <component_ref component="sodium_current_m_gate"/>
      </component_ref>
   </group>
```
(the second one is not allowed for 2 reasons, so might raise an error somewhere else)

Reply:
No

Reply:
Ah, Asif cutting corners then :)

Reply:
I see ok test added now